submission/file-sources/src/main/kotlin/uk/ac/ebi/io/sources/FtpSource.kt

```diff
@@ -7,13 +7,12 @@ import ebi.ac.uk.model.Attribute
 import ebi.ac.uk.model.constants.FileFields
 import uk.ac.ebi.io.builder.createFile
 import java.io.File
-import java.nio.file.Files
 import java.nio.file.Path
-import kotlin.io.path.outputStream
 
 class FtpSource(
     override val description: String,
-    private val basePath: Path,
+    private val ftpPath: Path,
```

Comment:
I think the names `ftpUrl` and `nfsFtpPath` would be clearer in order to differentiate which resource we're referring to